pkg/server/nodemanager/node_scaler.go

```diff
@@ -57,6 +57,15 @@ func (s *BindingNodeScaler) spotMatches(pod *api.Pod, node *api.Node) bool {
 	return false
 }
 
+// we try test to see if Dedicated is set true for pod.Spec.Dedicated and
+// ensure that pod.Spec.Spot is not also set to true.
+func (s *BindingNodeScaler) dedicatedMatches(pod *api.Pod, node *api.Node) bool {
+	if node.Spec.Dedicated && pod.Spec.Spot.Policy == api.SpotNever {
+		return true
+	}
+	return false
+}
```

Comment:
Is this function used anywhere?

Reply:
If it's used, it can be onliner: `return node.Spec.Dedicated && pod.Spec.Spot.Policy == api.SpotNever`

Reply:
This function should have been placed inside of the `podMatchesNode()` function on line 67 of node_scaler.go

Reply:
this has been simplified to ensure that pod.Spec.Dedicated matches node.Spec.Dedicated it has also been added to `podMatchesNode()` function where it was supposed to be called in  the first place.